### PR TITLE
completed MySql provider

### PR DESCRIPTION
**Pull request by mediocreguy**
The provider passes all integration tests - support could not be coded for RenameColumn() or RenamePrimaryKey().  Thanks for reviewing.

### CODE REVIEW

**Comment by mediocreguy** [comment]
OK - sorry - do you want me to fix?

**Comment by dradovic** [comment]
Great work! I have a couple of minor comments. Can you please have a look at these? Then I'll merge and then I can take over.

**Comment by mediocreguy** [comment]
Should be done - thanks for reviewing!

**Comment by dradovic** [comment]
Excellent. Thank you very much. I will double check everything, make a CI config for MySql, adapt the documentation and then MySql support will be part of the next release. Great!

File: MigSharp/MigSharp.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (dradovic) [inline MigSharp/MigSharp.csproj]: I wonder why the BOM has disappeared?

**Comment by dradovic** [comment]
I've just installed MySql on my machine to test everything. Unfortunately, Migration8 is failing. I have to considerably lower the sizes of the VARCHAR columns used (i.e. instead of using `column.OfSize(support.MaximumSize / 3)` I had to `/ 10` and PK of 200 works but no 300 or your 767). Do you have any idea what is going on? Do I have a special unicode enconding on which you don't have?

**Comment by mediocreguy** [comment]
I think you are correct - utf8 can take up to 3 bytes per character, so that would reduce the allowed row and PK size.

https://dev.mysql.com/doc/refman/5.0/en/column-count-limit.html

Your sizes are probably safer for testing.

**Comment by dradovic** [comment]
You're right. For the PK `maximumSize = 767 / 3` works while `maximumSize = 767 / 3 + 1` does not.

But now I spotted another funny thing. Still Mig8. If instead of `column.OfSize(support.MaximumSize / 3)` I do a `column.OfSize(support.MaximumSize)` or completely comment out that block, I works! Any explanation for this? Does my MySQL server version automatically swap out too long VARCHAR columns?

**Comment by mediocreguy** [comment]
Sorry for the slow response...

It looks like this has to do with MySQL's "strict mode" - when not enabled (which is the default I think) the server seems to do all sorts of magical things....

http://dev.mysql.com/doc/refman/5.5/en/sql-mode.html#sql-mode-strict

I know that INSERTS with certain errors will not fail and as you have pointed out VARCHAR columns are magically changed to MEDIUMTEXT columns in certain cases.

The following batch will fail - strict mode is turned on - but will pass if strict mode is left off (by default)

set sql_mode = "ansi_quotes";
set sql_mode = "strict_all_tables";
CREATE TABLE `Mig8`(
	`Column1` VARCHAR(65535)  NULL ,
	`Column2` VARCHAR(65535)  NULL ,
	`Column3` NUMERIC(28, 28)  NULL ,
	`Column4` LONGTEXT  NULL ,
	`Column5` TINYINT UNSIGNED  NULL ,
	`Column6` BIT  NULL ,
	`Column7` DATETIME  NULL ,
	`Column8` DOUBLE  NULL ,
	`Column9` SMALLINT  NULL ,
	`Column10` INT  NULL ,
	`Column11` FLOAT  NULL ,
	`Column12` LONGTEXT  NULL ,
	`Column13` SMALLINT UNSIGNED  NULL ,
	`Column14` INTEGER UNSIGNED  NULL ,
	`Column15` BIGINT UNSIGNED  NULL ) character set='utf8';
INSERT INTO "Mig8" ("Column1", "Column2", "Column3", "Column4", "Column5", "Column6", "Column7", "Column8", "Column9", "Column10", "Column11", "Column12", "Column13", "Column14", "Column15") VALUES ('Test', 'IrgendÃ¶ppis', 0.12345, 'Test', 255, 1, '2010-12-28T18:14:33', 3.14159265358979, 32767, 2147483647, 2.71828, 'IrgendÃ¶ppis', 0, 0, 0);

The current test - in UTF8 character set mode - does not effectively test the VARCHAR type.

I would like to suggest a couple of things:

1. For testing, turn on UTF8 for the test database - use "CREATE DATABASE `{0}` CHARACTER SET utf8 COLLATE utf8_general_ci;" in MySqlIntegrationTests.Setup()   My test server is setup with latin1 (unknown to me) - hence the difference in our results.

2. In Migration8, reduce the primary key size from 767 to 250 - due to UTF8

3. Also in Migration8, size AnsiString and String types to 5000 - this results in VARCHAR(5000) fields

 All tests pass with these updates.

**Comment by dradovic** [comment]
I've finally finished integrating the MySQL support. You can have a look at the result in my latest commit. Fundamentally, I didn't change a lot except I added support for the CurrentDateTime default.

**Comment by mediocreguy** [comment]
Looks great - Thanks for incorporating MySql support into MigSharp!

**Pull request merged**